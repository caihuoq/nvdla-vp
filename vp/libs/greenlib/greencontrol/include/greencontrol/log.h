//   GreenControl framework
//
// LICENSETEXT
//
//   Copyright (C) 2009 : GreenSocs Ltd
// 	 http://www.greensocs.com/ , email: info@greensocs.com
//
//   Developed by :
//   
//   Michael Ruetz <m.ruetz@tu-bs.de>,
//   Christian Schroeder <schroeder@eis.cs.tu-bs.de>
//     Technical University of Braunschweig, Dept. E.I.S.
//     http://www.eis.cs.tu-bs.de
//
//
//   This program is free software.
// 
//   If you have no applicable agreement with GreenSocs Ltd, this software
//   is licensed to you, and you can redistribute it and/or modify
//   it under the terms of the GNU General Public License as published by
//   the Free Software Foundation; either version 2 of the License, or
//   (at your option) any later version.
// 
//   If you have a applicable agreement with GreenSocs Ltd, the terms of that
//   agreement prevail.
// 
//   This program is distributed in the hope that it will be useful,
//   but WITHOUT ANY WARRANTY; without even the implied warranty of
//   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//   GNU General Public License for more details.
// 
//   You should have received a copy of the GNU General Public License
//   along with this program; if not, write to the Free Software
//   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
//   02110-1301  USA 
// 
// ENDLICENSETEXT

// doxygen comments

#ifndef __LOG_H__
#define __LOG_H__

#include "greencontrol/gclog/gc_logger.h"
#include "greencontrol/gclog/filter.h"
#include "greencontrol/gclog/output.h"

#endif
