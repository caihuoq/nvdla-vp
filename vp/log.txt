-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for NVDLA CMOD
-- NVDLA HW VERSION = nvdla_os_initial
-- NVDLA CMOD include directory = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/include
-- NVDLA CMOD library directory = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/lib
-- NVDLA CMOD library = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/lib/libnvdla_cmod.so
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for GreenLib
-- GreenLib include directory = /workspace/vp/libs/greenlib/greencontrol/include
-- GreenLib library directory = /workspace/vp/libs/greenlib/greenreg/lib
-- Searching for TLM2C
-- TLM2C include directory = /workspace/vp/libs/tlm2c/include
-- TLM2C library directory = /workspace/vp/libs/tlm2c.build
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for GreenLib
-- GreenLib include directory = /workspace/vp/libs/greenlib/greencontrol/include
-- GreenLib library directory = /workspace/vp/libs/greenlib/greenreg/lib
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for GreenLib
-- GreenLib include directory = /workspace/vp/libs/greenlib/greencontrol/include
-- GreenLib library directory = /workspace/vp/libs/greenlib/greenreg/lib
-- Searching for NVDLA CMOD
-- NVDLA HW VERSION = nvdla_os_initial
-- NVDLA CMOD include directory = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/include
-- NVDLA CMOD library directory = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/lib
-- NVDLA CMOD library = /usr/local/nvdla/hw/outdir/nv_full/cmod/release/lib/libnvdla_cmod.so
-- Searching for SystemC
-- SystemC version = 2.3.0
-- SystemC library = /usr/local/systemc-2.3.0/lib-linux64/libsystemc.so
-- Searching for TLM
running ls /usr/local/systemc-2.3.0/include/tlm.h 2>&1
/usr/local/systemc-2.3.0/include/tlm.h
-- TLM library = /usr/local/systemc-2.3.0/include/tlm.h
-- Searching for GreenLib
-- GreenLib include directory = /workspace/vp/libs/greenlib/greencontrol/include
-- GreenLib library directory = /workspace/vp/libs/greenlib/greenreg/lib
-- Searching for SimpleCPU model.
-- SimpleCPU include directory = /workspace/vp/models/cpu/include
-- SimpleCPU library directory = /workspace/vp/models/cpu.build
-- Configuring done
-- Generating done
-- Build files have been written to: /workspace/vp
