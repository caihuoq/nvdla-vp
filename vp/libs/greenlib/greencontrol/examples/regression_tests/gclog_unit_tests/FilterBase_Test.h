//   GreenControl framework
//
// LICENSETEXT
//
//   Copyright (C) 2009 : GreenSocs Ltd
// 	 http://www.greensocs.com/ , email: info@greensocs.com
//
//   Developed by :
//   
//   Michael Ruetz <m.ruetz@tu-bs.de>,
//   Christian Schroeder <schroeder@eis.cs.tu-bs.de>
//     Technical University of Braunschweig, Dept. E.I.S.
//     http://www.eis.cs.tu-bs.de
//
//
// The contents of this file are subject to the licensing terms specified
// in the file LICENSE. Please consult this file for restrictions and
// limitations that may apply.
// 
// ENDLICENSETEXT

// doxygen comments

#ifndef __FILTERBASE_TEST_H__
#define __FILTERBASE_TEST_H__

#include "greencontrol/log.h"
#include "TestCase.h"
#include "Filter_dummy.h"

// GCLOG_RETURN should be enabled
class FilterBase_Test : public TestCase
{
public:
  FilterBase_Test() {}
  virtual ~FilterBase_Test(void) {}

  std::string identify()
  {
    return "FilterBase_Test";
  }

  void setup()
  {
    pLog = new GC_Logger();
  }

  void destroy()
  {
    delete pLog;
  }

  // This is for testing the FilterBase and ServiceFilter classes
  void run()
  {
    DebugTextfile textOut("FilterBase_Test.txt", false);
    ControlTransactionHandle ct;

    // These Filter_dummys are being used to test the automatic creation of the filter ports.
    // However, the ports are being created when a filter is added to the dummy. So there are no warnings here.
    Filter_dummy invalidFilter(-1), noPortFilter(0), f(1), f2(100), toHigh(101);

    ct = getCT();

    // Test 1: Trying to create filters with an invalid number of filter ports
    textOut.writePure("Test 1\nTrying to create filters with an invalid number of filter ports.\n");

    // trying to add a NULL pointer and creating a filter with a negative number of filter ports
    std::cout << "\nA warning about creating a filter with too many filter ports should follow.\n";
    std::cout << "That warning should be followed by another warning about adding a NULL pointer:\n";
    invalidFilter.addFilter(0, NULL);

    // trying to create too many filter ports, the activate() callback of invalidFilter should be called
    std::cout << "\nA warning about creating a filter with too many filter ports should follow.\n";
    std::cout << "That warning should be followed by a message about activate() callback of the Filter_dummy.\n";
    toHigh.addFilter(0, &invalidFilter);

    // f doesn't have more than one filter port (index 0)
    std::cout << "\nA warning about a non existing filter port should follow:\n";
    f.addFilter(1, &toHigh);

    // the activate callback must not be called a second time
    std::cout << "\nThere should be no message about the activate() callback until the next test.\n";
    f.addFilter(0, &invalidFilter);

    // this filter has no filter ports
    std::cout << "\nA warning about a non existing filter port should follow:\n";
    noPortFilter.addFilter(0, &invalidFilter);

    // no warning, just add filters
    f2.addFilter(0, &invalidFilter);
    f2.addFilter(2, &textOut);

    // trying to use a non existing filter port
    std::cout << "\nA warning about a non existing filter port should follow:\n";
    noPortFilter.useFilterPort(0, ct);

    // using filter ports, but an output filter is attached to filter port 2 only
    textOut.writePure("Only one transaction should be logged.");
    f2.useFilterPort(0, ct);
    f2.useFilterPort(1, ct);
    f2.useFilterPort(2, ct);

    // Test 2: Testing the ServiceFilter.
    textOut.writePure("\nTest 2\nTesting the ServiceFilter with 2 filter ports.\n");

    ServiceFilter sFilt(2);

    std::cout << "\nA warning about a non existing filter port should follow:\n";
    sFilt.addFilter(2, &invalidFilter);
    sFilt.addFilter(0, &textOut);
    sFilt.addFilter(1, &textOut);

    // the AV_SERVICE CT should not be logged
    textOut.writePure("\nThis should be followed by two transactions, one NO_SERVICE CT and one CONFIG_SERVICE CT.\n");
    sFilt.processTransaction(ct);
    ct->set_mService(AV_SERVICE);
    sFilt.processTransaction(ct);
    ct->set_mService(CONFIG_SERVICE);
    sFilt.processTransaction(ct);
  }

protected:
  GC_Logger* pLog;
};

#endif
