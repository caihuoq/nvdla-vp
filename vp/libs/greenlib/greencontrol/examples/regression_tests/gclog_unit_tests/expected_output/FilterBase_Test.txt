Tuesday, 16 February 2010 - 13:06:06
Debug text output used with GC_Logger.
Starting log.

Test 1
Trying to create filters with an invalid number of filter ports.

Only one transaction should be logged.
  Transaction 0x93a24f8 returning:
    mHasChanged      = 1
    mService     (1) = 0 (NO_SERVICE)
    mTarget      (1) = 0
    mCmd         (1) = 0 (Warning: API/Plugin doesn't implement command_if!)
    mAnyPointer  (1) = 0
    mAnyPointer2 (1) = 0
    mAnyUint     (1) = 64
    mSpecifier   (1) = 
    mValue       (1) = 
    mID          (1) = 0x93a16a8 (Warning: API/Plugin doesn't implement command_if!)
    mMetaData    (1) = 
    mError       (1) = 0
    mLogPointer  (1) = 0
    mCmdIf       (1) = 0
    mLogConfig   (1) = 0x93a2538 (hasChanged: 1)
    -> pNext     (0) = 0

Test 2
Testing the ServiceFilter with 2 filter ports.


This should be followed by two transactions, one NO_SERVICE CT and one CONFIG_SERVICE CT.

  Transaction 0x93a24f8 returning:
    mHasChanged      = 1
    mService     (1) = 0 (NO_SERVICE)
    mTarget      (1) = 0
    mCmd         (1) = 0 (Warning: API/Plugin doesn't implement command_if!)
    mAnyPointer  (1) = 0
    mAnyPointer2 (1) = 0
    mAnyUint     (1) = 64
    mSpecifier   (1) = 
    mValue       (1) = 
    mID          (1) = 0x93a16a8 (Warning: API/Plugin doesn't implement command_if!)
    mMetaData    (1) = 
    mError       (1) = 0
    mLogPointer  (1) = 0
    mCmdIf       (1) = 0
    mLogConfig   (1) = 0x93a2538 (hasChanged: 1)
    -> pNext     (0) = 0
  Transaction 0x93a24f8 returning:
    mHasChanged      = 1
    mService     (1) = 1 (CONFIG_SERVICE)
    mTarget      (1) = 0
    mCmd         (1) = 0 (Warning: API/Plugin doesn't implement command_if!)
    mAnyPointer  (1) = 0
    mAnyPointer2 (1) = 0
    mAnyUint     (1) = 64
    mSpecifier   (1) = 
    mValue       (1) = 
    mID          (1) = 0x93a16a8 (Warning: API/Plugin doesn't implement command_if!)
    mMetaData    (1) = 
    mError       (1) = 0
    mLogPointer  (1) = 0
    mCmdIf       (1) = 0
    mLogConfig   (1) = 0x93a2538 (hasChanged: 1)
    -> pNext     (0) = 0

Regular end of log.
Finished on Tuesday, 16 February 2010 - 13:06:06
Closing log.